cmake/OpenCVCompilerOptimizations.cmake

```diff
@@ -380,6 +380,10 @@ if(CV_DISABLE_OPTIMIZATION)
   set(CPU_DISPATCH_REQUIRE "")
 endif()
 
+if("x${CPU_DISPATCH}" STREQUAL "xALL")
```

Comment:
It would be better to if we add an option to exclude specific optimization,
something like that 
usage  `-DCPU_DISPATCH=ALL-AVX512_SKX-SSE4_1`
```CMAKE
if("${CPU_DISPATCH}" MATCHES "^(ALL)")
  string(REPLACE "-" ";" _exclude_opt ${CPU_DISPATCH})
  set(CPU_DISPATCH "${CPU_KNOWN_OPTIMIZATIONS}")
  foreach(_exopt ${_exclude_opt})
    list(REMOVE_ITEM CPU_DISPATCH ${_exopt})
  endforeach()
endif()
````

Reply:
I prefer to store excluded entries in a separate variables, like [`CPU_BASELINE_DISABLE`](https://github.com/opencv/opencv/blob/3.4.6/cmake/OpenCVCompilerOptimizations.cmake#L24-L26) - due better flexibility.

Reply:
excluded entries in a separate variable like `CPU_BASELINE_DISABLE` is reasonable since we force implies baseline opts but when it comes to dispatch `ALL` with unary minus and also `DEFAULT`! with unary addition seems more practical to me

Reply:
@seiko2plus That makes sense.

Actually real use cases of excluding from ALL is still not clear to me (except some debugging).

Main purpose of `ALL` value is enable testing of all available CPU configurations (even if they are not in the default list).

Anyway feel free to prepare separate PR for that.